protocols/kad/src/handler.rs

```diff
@@ -768,16 +754,6 @@ impl ConnectionHandler for Handler {
             });
         }
 
-        let no_streams = self.outbound_substreams.is_empty() && self.inbound_substreams.is_empty();
```

Comment:
For my own reference, this also removes the check for inbound and outbound streams, meaning we MUST NOT merge this before https://github.com/libp2p/rust-libp2p/pull/4595.

Reply:
@leonzchang Can you restore the check here in regards to `inbound_streams` and `outbound_streams`? Then we can merge this independently and clean up all of this in #4595!

Reply:
sure, fixed 0e4036a

Reply:
Awesome! We could even make it a bit easier by just doing this computation in `connection_keep_alive` now because we don't depend on our prior state of `KeepAlive` anymore.

In other words, remove the field and just compute it on demand.

Reply:
Cool, much cleaner 909adf0

Reply:
> @leonzchang Can you restore the check here in regards to `inbound_streams` and `outbound_streams`? Then we can merge this independently and clean up all of this in #4595!

I will do the clean up in #4595 later.